src/FACT_internal.c

```diff
@@ -1648,6 +1648,8 @@ void FACT_INTERNAL_OnBufferEnd(FAudioVoiceCallback *callback, void* pContext)
 	if (	(c->wave->streamOffset >= end) ||
 		(c->wave->state & FACT_STATE_STOPPED)	)
 	{
+		/* We shouldn't ever get here anymore */
```

Comment:
This comment can be turned into an assert message.

Reply:
Upon further review it looks like we will get here due to the XAudio2 spec. Has to be undone with the FAudio_internal changes.